.github/workflows/ci-hardening.yml

```
@@ -0,0 +1,158 @@
+      - name: Discover Dockerfiles
+        id: df
+        run: |
+          set -euo pipefail
+          mapfile -t FILES < <(git ls-files | grep -E '^docker/.*Dockerfile|Dockerfile$|Dockerfile\.' || true)
+          if [ ${#FILES[@]} -eq 0 ]; then
+            # Fallback to base non-root hardening image to keep pipeline green
+            FILES=(docker/base/Dockerfile.nonroot)
+          fi
+          printf "%s\n" "${FILES[@]}" | sed 's|^|file=|g' >> $GITHUB_OUTPUT
+      - name: Build images
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:9:18: Double quote to prevent globbing and word splitting

(shellcheck)